Index: SuperLU_3.0/TESTING/ddrive.c
===================================================================
--- SuperLU_3.0.orig/TESTING/ddrive.c
+++ SuperLU_3.0/TESTING/ddrive.c
@@ -72,9 +72,9 @@ main(int argc, char *argv[])
     fact_t         fact;
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
     static char    equeds[]  = {'N', 'R', 'C', 'B'};
Index: SuperLU_3.0/TESTING/sdrive.c
===================================================================
--- SuperLU_3.0.orig/TESTING/sdrive.c
+++ SuperLU_3.0/TESTING/sdrive.c
@@ -72,9 +72,9 @@ main(int argc, char *argv[])
     fact_t         fact;
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
     static char    equeds[]  = {'N', 'R', 'C', 'B'};
Index: SuperLU_3.0/TESTING/zdrive.c
===================================================================
--- SuperLU_3.0.orig/TESTING/zdrive.c
+++ SuperLU_3.0/TESTING/zdrive.c
@@ -72,9 +72,9 @@ main(int argc, char *argv[])
     fact_t         fact;
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
     static char    equeds[]  = {'N', 'R', 'C', 'B'};
Index: SuperLU_3.0/TESTING/cdrive.c
===================================================================
--- SuperLU_3.0.orig/TESTING/cdrive.c
+++ SuperLU_3.0/TESTING/cdrive.c
@@ -72,9 +72,9 @@ main(int argc, char *argv[])
     fact_t         fact;
     trans_t        trans;
     SuperLUStat_t  stat;
     static char    matrix_type[8];
-    static char    equed[1], path[3], sym[1], dist[1];
+    static char    equed[1], path[4], sym[1], dist[1];
 
     /* Fixed set of parameters */
     int            iseed[]  = {1988, 1989, 1990, 1991};
     static char    equeds[]  = {'N', 'R', 'C', 'B'};
