--- Makefile.old	2007-11-10 21:01:12.000000000 -0500
+++ Makefile	2007-11-10 21:05:45.000000000 -0500
@@ -14,6 +14,8 @@
 #		    October 15, 2003 Version 3.0
 #
 ############################################################################
+DESTDIR =
+DESTLIBDIR = $(DESTDIR)/$(LIBDIR)
 
 include make.inc
 
@@ -54,3 +56,6 @@
 	( cd MATLAB;  $(MAKE) clean )
 	( cd EXAMPLE; $(MAKE) clean )
 	( cd FORTRAN; $(MAKE) clean )
+ 
+
+
