diff -Naur SuperLU_5.2.1/SRC/Makefile SuperLU_5.2.1.tpg/SRC/Makefile
--- SuperLU_5.2.1/SRC/Makefile	2016-05-22 15:58:44.000000000 +0000
+++ SuperLU_5.2.1.tpg/SRC/Makefile	2017-12-11 18:31:28.147024684 +0000
@@ -98,7 +98,7 @@
 	ilu_zcolumn_dfs.o ilu_zpanel_dfs.o ilu_zcopy_to_ucol.o \
 	ilu_zpivotL.o zdiagonal.o zlacon2.o dzsum1.o izmax1.o
 
-all:    single double complex complex16
+all:    sharedlib
 
 single: $(SLUSRC) $(ALLAUX) $(SCAUX)
 	$(ARCH) $(ARCHFLAGS) $(SUPERLULIB) $(SLUSRC) $(ALLAUX) $(SCAUX)
@@ -116,10 +116,16 @@
 	$(ARCH) $(ARCHFLAGS) $(SUPERLULIB) $(ZLUSRC) $(ALLAUX) $(DZLAUX)
 	$(RANLIB) $(SUPERLULIB)
 
+sharedlib: $(ALLAUX) $(LAAUX) $(SLASRC) $(DLASRC) $(CLASRC) $(ZLASRC) $(SCLAUX) $(DZLAUX) $(SLUSRC) $(DLUSRC) $(CLUSRC) $(ZLUSRC)
+	$(CC) $(CFLAGS) $(LIBS) -shared -Wl,-soname,libsuperlu.so.5.2 -o libsuperlu.so.5.2 \
+	$(ALLAUX) $(LAAUX) $(SLASRC) $(DLASRC) $(CLASRC) $(ZLASRC) $(SCLAUX) \
+	$(DZLAUX) $(SLUSRC) $(DLUSRC) $(CLUSRC) $(ZLUSRC)
+	ln -sf libsuperlu.so.5.2 libsuperlu.so
+
 ##################################
 
 .c.o:
-	$(CC) $(CFLAGS) $(CDEFS) $(BLASDEF) -c $< $(VERBOSE)
+	$(CC) $(CFLAGS) $(CDEFS) $(BLASDEF) $(LIBS) -c $< $(VERBOSE)
 
 .f.o:
 	$(FORTRAN) $(FFLAGS) -c $<
