Index: SuperLU_3.0/CBLAS/cgemv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/cgemv.c
+++ SuperLU_3.0/CBLAS/cgemv.c
@@ -12,9 +12,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     complex q__1, q__2, q__3;
 
     /* Builtin functions */
     void r_cnjg(complex *, complex *);
@@ -194,19 +193,14 @@
 
     if (beta->r != 1.f || beta->i != 0.f) {
 	if (*incy == 1) {
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0.f, Y(i).i = 0.f;
 /* L10: */
 		}
 	    } else {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    q__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    q__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
@@ -215,20 +209,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0.f, Y(iy).i = 0.f;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    q__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    q__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
@@ -246,22 +235,15 @@
 /*        Form  y := alpha*A*x + y. */
 
 	jx = kx;
 	if (*incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		if (X(jx).r != 0.f || X(jx).i != 0.f) {
-		    i__2 = jx;
 		    q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    q__1.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    temp.r = q__1.r, temp.i = q__1.i;
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i;
-			i__4 = i;
-			i__5 = i + j * a_dim1;
 			q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				q__2.i = temp.r * A(i,j).i + temp.i * A(i,j)
 				.r;
 			q__1.r = Y(i).r + q__2.r, q__1.i = Y(i).i + 
@@ -273,23 +255,16 @@
 		jx += *incx;
 /* L60: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		if (X(jx).r != 0.f || X(jx).i != 0.f) {
-		    i__2 = jx;
 		    q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    q__1.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    temp.r = q__1.r, temp.i = q__1.i;
 		    iy = ky;
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = iy;
-			i__4 = iy;
-			i__5 = i + j * a_dim1;
 			q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				q__2.i = temp.r * A(i,j).i + temp.i * A(i,j)
 				.r;
 			q__1.r = Y(iy).r + q__2.r, q__1.i = Y(iy).i + 
@@ -309,55 +284,43 @@
  */
 
 	jy = ky;
 	if (*incx == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
 		temp.r = 0.f, temp.i = 0.f;
 		if (noconj) {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i;
 			q__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i)
 				.i, q__2.i = A(i,j).r * X(i).i + A(i,j)
 				.i * X(i).r;
 			q__1.r = temp.r + q__2.r, q__1.i = temp.i + q__2.i;
 			temp.r = q__1.r, temp.i = q__1.i;
 /* L90: */
 		    }
 		} else {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
 			r_cnjg(&q__3, &A(i,j));
-			i__3 = i;
 			q__2.r = q__3.r * X(i).r - q__3.i * X(i).i, 
 				q__2.i = q__3.r * X(i).i + q__3.i * X(i)
 				.r;
 			q__1.r = temp.r + q__2.r, q__1.i = temp.i + q__2.i;
 			temp.r = q__1.r, temp.i = q__1.i;
 /* L100: */
 		    }
 		}
-		i__2 = jy;
-		i__3 = jy;
 		q__2.r = alpha->r * temp.r - alpha->i * temp.i, q__2.i = 
 			alpha->r * temp.i + alpha->i * temp.r;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
 		jy += *incy;
 /* L110: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
 		temp.r = 0.f, temp.i = 0.f;
 		ix = kx;
 		if (noconj) {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = ix;
 			q__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix)
 				.i, q__2.i = A(i,j).r * X(ix).i + A(i,j)
 				.i * X(ix).r;
 			q__1.r = temp.r + q__2.r, q__1.i = temp.i + q__2.i;
@@ -365,12 +328,10 @@
 			ix += *incx;
 /* L120: */
 		    }
 		} else {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
 			r_cnjg(&q__3, &A(i,j));
-			i__3 = ix;
 			q__2.r = q__3.r * X(ix).r - q__3.i * X(ix).i, 
 				q__2.i = q__3.r * X(ix).i + q__3.i * X(ix)
 				.r;
 			q__1.r = temp.r + q__2.r, q__1.i = temp.i + q__2.i;
@@ -378,10 +339,8 @@
 			ix += *incx;
 /* L130: */
 		    }
 		}
-		i__2 = jy;
-		i__3 = jy;
 		q__2.r = alpha->r * temp.r - alpha->i * temp.i, q__2.i = 
 			alpha->r * temp.i + alpha->i * temp.r;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
Index: SuperLU_3.0/CBLAS/cgerc.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/cgerc.c
+++ SuperLU_3.0/CBLAS/cgerc.c
@@ -11,9 +11,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     complex q__1, q__2;
 
     /* Builtin functions */
     void r_cnjg(complex *, complex *);
@@ -140,21 +139,15 @@
     } else {
 	jy = 1 - (*n - 1) * *incy;
     }
     if (*incx == 1) {
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = jy;
 	    if (Y(jy).r != 0.f || Y(jy).i != 0.f) {
 		r_cnjg(&q__2, &Y(jy));
 		q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			alpha->r * q__2.i + alpha->i * q__2.r;
 		temp.r = q__1.r, temp.i = q__1.i;
-		i__2 = *m;
 		for (i = 1; i <= *m; ++i) {
-		    i__3 = i + j * a_dim1;
-		    i__4 = i + j * a_dim1;
-		    i__5 = i;
 		    q__2.r = X(i).r * temp.r - X(i).i * temp.i, q__2.i =
 			     X(i).r * temp.i + X(i).i * temp.r;
 		    q__1.r = A(i,j).r + q__2.r, q__1.i = A(i,j).i + q__2.i;
 		    A(i,j).r = q__1.r, A(i,j).i = q__1.i;
@@ -169,22 +162,16 @@
 	    kx = 1;
 	} else {
 	    kx = 1 - (*m - 1) * *incx;
 	}
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = jy;
 	    if (Y(jy).r != 0.f || Y(jy).i != 0.f) {
 		r_cnjg(&q__2, &Y(jy));
 		q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			alpha->r * q__2.i + alpha->i * q__2.r;
 		temp.r = q__1.r, temp.i = q__1.i;
 		ix = kx;
-		i__2 = *m;
 		for (i = 1; i <= *m; ++i) {
-		    i__3 = i + j * a_dim1;
-		    i__4 = i + j * a_dim1;
-		    i__5 = ix;
 		    q__2.r = X(ix).r * temp.r - X(ix).i * temp.i, q__2.i =
 			     X(ix).r * temp.i + X(ix).i * temp.r;
 		    q__1.r = A(i,j).r + q__2.r, q__1.i = A(i,j).i + q__2.i;
 		    A(i,j).r = q__1.r, A(i,j).i = q__1.i;
Index: SuperLU_3.0/CBLAS/chemv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/chemv.c
+++ SuperLU_3.0/CBLAS/chemv.c
@@ -12,9 +12,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublereal d__1;
     complex q__1, q__2, q__3, q__4;
 
     /* Builtin functions */
@@ -180,19 +179,14 @@
 
     if (beta->r != 1.f || beta->i != 0.f) {
 	if (*incy == 1) {
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0.f, Y(i).i = 0.f;
 /* L10: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    q__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    q__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
@@ -201,20 +195,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0.f, Y(iy).i = 0.f;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    q__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    q__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
@@ -231,36 +220,26 @@
 
 /*        Form  y  when A is stored in upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		q__1.r = alpha->r * X(j).r - alpha->i * X(j).i, q__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(i).r + q__2.r, q__1.i = Y(i).i + q__2.i;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
 		    r_cnjg(&q__3, &A(i,j));
-		    i__3 = i;
 		    q__2.r = q__3.r * X(i).r - q__3.i * X(i).i, q__2.i =
 			     q__3.r * X(i).i + q__3.i * X(i).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L50: */
 		}
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		q__3.r = d__1 * temp1.r, q__3.i = d__1 * temp1.i;
 		q__2.r = Y(j).r + q__3.r, q__2.i = Y(j).i + q__3.i;
 		q__4.r = alpha->r * temp2.r - alpha->i * temp2.i, q__4.i = 
@@ -271,40 +250,30 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, q__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
 		ix = kx;
 		iy = ky;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(iy).r + q__2.r, q__1.i = Y(iy).i + q__2.i;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
 		    r_cnjg(&q__3, &A(i,j));
-		    i__3 = ix;
 		    q__2.r = q__3.r * X(ix).r - q__3.i * X(ix).i, q__2.i =
 			     q__3.r * X(ix).i + q__3.i * X(ix).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 		    ix += *incx;
 		    iy += *incy;
 /* L70: */
 		}
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		q__3.r = d__1 * temp1.r, q__3.i = d__1 * temp1.i;
 		q__2.r = Y(jy).r + q__3.r, q__2.i = Y(jy).i + q__3.i;
 		q__4.r = alpha->r * temp2.r - alpha->i * temp2.i, q__4.i = 
@@ -320,42 +289,30 @@
 
 /*        Form  y  when A is stored in lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		q__1.r = alpha->r * X(j).r - alpha->i * X(j).i, q__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		q__2.r = d__1 * temp1.r, q__2.i = d__1 * temp1.i;
 		q__1.r = Y(j).r + q__2.r, q__1.i = Y(j).i + q__2.i;
 		Y(j).r = q__1.r, Y(j).i = q__1.i;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(i).r + q__2.r, q__1.i = Y(i).i + q__2.i;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
 		    r_cnjg(&q__3, &A(i,j));
-		    i__3 = i;
 		    q__2.r = q__3.r * X(i).r - q__3.i * X(i).i, q__2.i =
 			     q__3.r * X(i).i + q__3.i * X(i).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L90: */
 		}
-		i__2 = j;
-		i__3 = j;
 		q__2.r = alpha->r * temp2.r - alpha->i * temp2.i, q__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		q__1.r = Y(j).r + q__2.r, q__1.i = Y(j).i + q__2.i;
 		Y(j).r = q__1.r, Y(j).i = q__1.i;
@@ -363,46 +320,34 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, q__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		q__2.r = d__1 * temp1.r, q__2.i = d__1 * temp1.i;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
 		ix = jx;
 		iy = jy;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
 		    ix += *incx;
 		    iy += *incy;
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(iy).r + q__2.r, q__1.i = Y(iy).i + q__2.i;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
 		    r_cnjg(&q__3, &A(i,j));
-		    i__3 = ix;
 		    q__2.r = q__3.r * X(ix).r - q__3.i * X(ix).i, q__2.i =
 			     q__3.r * X(ix).i + q__3.i * X(ix).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L110: */
 		}
-		i__2 = jy;
-		i__3 = jy;
 		q__2.r = alpha->r * temp2.r - alpha->i * temp2.i, q__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
Index: SuperLU_3.0/CBLAS/cher2.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/cher2.c
+++ SuperLU_3.0/CBLAS/cher2.c
@@ -11,9 +11,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5, i__6;
     doublereal d__1;
     complex q__1, q__2, q__3, q__4;
 
     /* Builtin functions */
@@ -184,93 +183,69 @@
 
 /*        Form  A  when A is stored in the upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
-		i__3 = j;
 		if (X(j).r != 0.f || X(j).i != 0.f || (Y(j).r != 0.f 
 			|| Y(j).i != 0.f)) {
 		    r_cnjg(&q__2, &Y(j));
 		    q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			    alpha->r * q__2.i + alpha->i * q__2.r;
 		    temp1.r = q__1.r, temp1.i = q__1.i;
-		    i__2 = j;
 		    q__2.r = alpha->r * X(j).r - alpha->i * X(j).i, 
 			    q__2.i = alpha->r * X(j).i + alpha->i * X(j)
 			    .r;
 		    r_cnjg(&q__1, &q__2);
 		    temp2.r = q__1.r, temp2.i = q__1.i;
-		    i__2 = j - 1;
 		    for (i = 1; i <= j-1; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = i;
 			q__3.r = X(i).r * temp1.r - X(i).i * temp1.i, 
 				q__3.i = X(i).r * temp1.i + X(i).i * 
 				temp1.r;
 			q__2.r = A(i,j).r + q__3.r, q__2.i = A(i,j).i + 
 				q__3.i;
-			i__6 = i;
 			q__4.r = Y(i).r * temp2.r - Y(i).i * temp2.i, 
 				q__4.i = Y(i).r * temp2.i + Y(i).i * 
 				temp2.r;
 			q__1.r = q__2.r + q__4.r, q__1.i = q__2.i + q__4.i;
 			A(i,j).r = q__1.r, A(i,j).i = q__1.i;
 /* L10: */
 		    }
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = j;
 		    q__2.r = X(j).r * temp1.r - X(j).i * temp1.i, 
 			    q__2.i = X(j).r * temp1.i + X(j).i * 
 			    temp1.r;
-		    i__5 = j;
 		    q__3.r = Y(j).r * temp2.r - Y(j).i * temp2.i, 
 			    q__3.i = Y(j).r * temp2.i + Y(j).i * 
 			    temp2.r;
 		    q__1.r = q__2.r + q__3.r, q__1.i = q__2.i + q__3.i;
 		    d__1 = A(j,j).r + q__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		}
 /* L20: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
-		i__3 = jy;
 		if (X(jx).r != 0.f || X(jx).i != 0.f || (Y(jy).r != 0.f 
 			|| Y(jy).i != 0.f)) {
 		    r_cnjg(&q__2, &Y(jy));
 		    q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			    alpha->r * q__2.i + alpha->i * q__2.r;
 		    temp1.r = q__1.r, temp1.i = q__1.i;
-		    i__2 = jx;
 		    q__2.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    q__2.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    r_cnjg(&q__1, &q__2);
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 		    ix = kx;
 		    iy = ky;
-		    i__2 = j - 1;
 		    for (i = 1; i <= j-1; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = ix;
 			q__3.r = X(ix).r * temp1.r - X(ix).i * temp1.i, 
 				q__3.i = X(ix).r * temp1.i + X(ix).i * 
 				temp1.r;
 			q__2.r = A(i,j).r + q__3.r, q__2.i = A(i,j).i + 
 				q__3.i;
-			i__6 = iy;
 			q__4.r = Y(iy).r * temp2.r - Y(iy).i * temp2.i, 
 				q__4.i = Y(iy).r * temp2.i + Y(iy).i * 
 				temp2.r;
 			q__1.r = q__2.r + q__4.r, q__1.i = q__2.i + q__4.i;
@@ -278,24 +253,18 @@
 			ix += *incx;
 			iy += *incy;
 /* L30: */
 		    }
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = jx;
 		    q__2.r = X(jx).r * temp1.r - X(jx).i * temp1.i, 
 			    q__2.i = X(jx).r * temp1.i + X(jx).i * 
 			    temp1.r;
-		    i__5 = jy;
 		    q__3.r = Y(jy).r * temp2.r - Y(jy).i * temp2.i, 
 			    q__3.i = Y(jy).r * temp2.i + Y(jy).i * 
 			    temp2.r;
 		    q__1.r = q__2.r + q__3.r, q__1.i = q__2.i + q__3.i;
 		    d__1 = A(j,j).r + q__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		}
 		jx += *incx;
@@ -307,118 +276,88 @@
 
 /*        Form  A  when A is stored in the lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
-		i__3 = j;
 		if (X(j).r != 0.f || X(j).i != 0.f || (Y(j).r != 0.f 
 			|| Y(j).i != 0.f)) {
 		    r_cnjg(&q__2, &Y(j));
 		    q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			    alpha->r * q__2.i + alpha->i * q__2.r;
 		    temp1.r = q__1.r, temp1.i = q__1.i;
-		    i__2 = j;
 		    q__2.r = alpha->r * X(j).r - alpha->i * X(j).i, 
 			    q__2.i = alpha->r * X(j).i + alpha->i * X(j)
 			    .r;
 		    r_cnjg(&q__1, &q__2);
 		    temp2.r = q__1.r, temp2.i = q__1.i;
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = j;
 		    q__2.r = X(j).r * temp1.r - X(j).i * temp1.i, 
 			    q__2.i = X(j).r * temp1.i + X(j).i * 
 			    temp1.r;
-		    i__5 = j;
 		    q__3.r = Y(j).r * temp2.r - Y(j).i * temp2.i, 
 			    q__3.i = Y(j).r * temp2.i + Y(j).i * 
 			    temp2.r;
 		    q__1.r = q__2.r + q__3.r, q__1.i = q__2.i + q__3.i;
 		    d__1 = A(j,j).r + q__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
-		    i__2 = *n;
 		    for (i = j + 1; i <= *n; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = i;
 			q__3.r = X(i).r * temp1.r - X(i).i * temp1.i, 
 				q__3.i = X(i).r * temp1.i + X(i).i * 
 				temp1.r;
 			q__2.r = A(i,j).r + q__3.r, q__2.i = A(i,j).i + 
 				q__3.i;
-			i__6 = i;
 			q__4.r = Y(i).r * temp2.r - Y(i).i * temp2.i, 
 				q__4.i = Y(i).r * temp2.i + Y(i).i * 
 				temp2.r;
 			q__1.r = q__2.r + q__4.r, q__1.i = q__2.i + q__4.i;
 			A(i,j).r = q__1.r, A(i,j).i = q__1.i;
 /* L50: */
 		    }
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		}
 /* L60: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
-		i__3 = jy;
 		if (X(jx).r != 0.f || X(jx).i != 0.f || (Y(jy).r != 0.f 
 			|| Y(jy).i != 0.f)) {
 		    r_cnjg(&q__2, &Y(jy));
 		    q__1.r = alpha->r * q__2.r - alpha->i * q__2.i, q__1.i = 
 			    alpha->r * q__2.i + alpha->i * q__2.r;
 		    temp1.r = q__1.r, temp1.i = q__1.i;
-		    i__2 = jx;
 		    q__2.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    q__2.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    r_cnjg(&q__1, &q__2);
 		    temp2.r = q__1.r, temp2.i = q__1.i;
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = jx;
 		    q__2.r = X(jx).r * temp1.r - X(jx).i * temp1.i, 
 			    q__2.i = X(jx).r * temp1.i + X(jx).i * 
 			    temp1.r;
-		    i__5 = jy;
 		    q__3.r = Y(jy).r * temp2.r - Y(jy).i * temp2.i, 
 			    q__3.i = Y(jy).r * temp2.i + Y(jy).i * 
 			    temp2.r;
 		    q__1.r = q__2.r + q__3.r, q__1.i = q__2.i + q__3.i;
 		    d__1 = A(j,j).r + q__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		    ix = jx;
 		    iy = jy;
-		    i__2 = *n;
 		    for (i = j + 1; i <= *n; ++i) {
 			ix += *incx;
 			iy += *incy;
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = ix;
 			q__3.r = X(ix).r * temp1.r - X(ix).i * temp1.i, 
 				q__3.i = X(ix).r * temp1.i + X(ix).i * 
 				temp1.r;
 			q__2.r = A(i,j).r + q__3.r, q__2.i = A(i,j).i + 
 				q__3.i;
-			i__6 = iy;
 			q__4.r = Y(iy).r * temp2.r - Y(iy).i * temp2.i, 
 				q__4.i = Y(iy).r * temp2.i + Y(iy).i * 
 				temp2.r;
 			q__1.r = q__2.r + q__4.r, q__1.i = q__2.i + q__4.i;
 			A(i,j).r = q__1.r, A(i,j).i = q__1.i;
 /* L70: */
 		    }
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.f;
 		}
 		jx += *incx;
Index: SuperLU_3.0/CBLAS/ctrsv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/ctrsv.c
+++ SuperLU_3.0/CBLAS/ctrsv.c
@@ -11,9 +11,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     complex q__1, q__2, q__3;
 
     /* Builtin functions */
     void c_div(complex *, complex *, complex *), r_cnjg(complex *, complex *);
@@ -186,21 +185,15 @@
 
 	if (lsame_(uplo, "U")) {
 	    if (*incx == 1) {
 		for (j = *n; j >= 1; --j) {
-		    i__1 = j;
 		    if (X(j).r != 0.f || X(j).i != 0.f) {
 			if (nounit) {
-			    i__1 = j;
 			    c_div(&q__1, &X(j), &A(j,j));
 			    X(j).r = q__1.r, X(j).i = q__1.i;
 			}
-			i__1 = j;
 			temp.r = X(j).r, temp.i = X(j).i;
 			for (i = j - 1; i >= 1; --i) {
-			    i__1 = i;
-			    i__2 = i;
-			    i__3 = i + j * a_dim1;
 			    q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    q__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    q__1.r = X(i).r - q__2.r, q__1.i = X(i).i - 
 				    q__2.i;
@@ -212,23 +205,17 @@
 		}
 	    } else {
 		jx = kx + (*n - 1) * *incx;
 		for (j = *n; j >= 1; --j) {
-		    i__1 = jx;
 		    if (X(jx).r != 0.f || X(jx).i != 0.f) {
 			if (nounit) {
-			    i__1 = jx;
 			    c_div(&q__1, &X(jx), &A(j,j));
 			    X(jx).r = q__1.r, X(jx).i = q__1.i;
 			}
-			i__1 = jx;
 			temp.r = X(jx).r, temp.i = X(jx).i;
 			ix = jx;
 			for (i = j - 1; i >= 1; --i) {
 			    ix -= *incx;
-			    i__1 = ix;
-			    i__2 = ix;
-			    i__3 = i + j * a_dim1;
 			    q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    q__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    q__1.r = X(ix).r - q__2.r, q__1.i = X(ix).i - 
 				    q__2.i;
@@ -241,24 +228,16 @@
 		}
 	    }
 	} else {
 	    if (*incx == 1) {
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = j;
 		    if (X(j).r != 0.f || X(j).i != 0.f) {
 			if (nounit) {
-			    i__2 = j;
 			    c_div(&q__1, &X(j), &A(j,j));
 			    X(j).r = q__1.r, X(j).i = q__1.i;
 			}
-			i__2 = j;
 			temp.r = X(j).r, temp.i = X(j).i;
-			i__2 = *n;
 			for (i = j + 1; i <= *n; ++i) {
-			    i__3 = i;
-			    i__4 = i;
-			    i__5 = i + j * a_dim1;
 			    q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    q__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    q__1.r = X(i).r - q__2.r, q__1.i = X(i).i - 
 				    q__2.i;
@@ -269,26 +248,18 @@
 /* L60: */
 		}
 	    } else {
 		jx = kx;
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = jx;
 		    if (X(jx).r != 0.f || X(jx).i != 0.f) {
 			if (nounit) {
-			    i__2 = jx;
 			    c_div(&q__1, &X(jx), &A(j,j));
 			    X(jx).r = q__1.r, X(jx).i = q__1.i;
 			}
-			i__2 = jx;
 			temp.r = X(jx).r, temp.i = X(jx).i;
 			ix = jx;
-			i__2 = *n;
 			for (i = j + 1; i <= *n; ++i) {
 			    ix += *incx;
-			    i__3 = ix;
-			    i__4 = ix;
-			    i__5 = i + j * a_dim1;
 			    q__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    q__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    q__1.r = X(ix).r - q__2.r, q__1.i = X(ix).i - 
 				    q__2.i;
@@ -306,17 +277,12 @@
 /*        Form  x := inv( A' )*x  or  x := inv( conjg( A' ) )*x. */
 
 	if (lsame_(uplo, "U")) {
 	    if (*incx == 1) {
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = j;
 		    temp.r = X(j).r, temp.i = X(j).i;
 		    if (noconj) {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
-			    i__3 = i + j * a_dim1;
-			    i__4 = i;
 			    q__2.r = A(i,j).r * X(i).r - A(i,j).i * X(
 				    i).i, q__2.i = A(i,j).r * X(i).i + 
 				    A(i,j).i * X(i).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -328,12 +294,10 @@
 			    c_div(&q__1, &temp, &A(j,j));
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    } else {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
 			    r_cnjg(&q__3, &A(i,j));
-			    i__3 = i;
 			    q__2.r = q__3.r * X(i).r - q__3.i * X(i).i, 
 				    q__2.i = q__3.r * X(i).i + q__3.i * X(
 				    i).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -346,24 +310,18 @@
 			    c_div(&q__1, &temp, &q__2);
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    }
-		    i__2 = j;
 		    X(j).r = temp.r, X(j).i = temp.i;
 /* L110: */
 		}
 	    } else {
 		jx = kx;
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
 		    ix = kx;
-		    i__2 = jx;
 		    temp.r = X(jx).r, temp.i = X(jx).i;
 		    if (noconj) {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
-			    i__3 = i + j * a_dim1;
-			    i__4 = ix;
 			    q__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(
 				    ix).i, q__2.i = A(i,j).r * X(ix).i + 
 				    A(i,j).i * X(ix).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -376,12 +334,10 @@
 			    c_div(&q__1, &temp, &A(j,j));
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    } else {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
 			    r_cnjg(&q__3, &A(i,j));
-			    i__3 = ix;
 			    q__2.r = q__3.r * X(ix).r - q__3.i * X(ix).i, 
 				    q__2.i = q__3.r * X(ix).i + q__3.i * X(
 				    ix).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -395,24 +351,19 @@
 			    c_div(&q__1, &temp, &q__2);
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    }
-		    i__2 = jx;
 		    X(jx).r = temp.r, X(jx).i = temp.i;
 		    jx += *incx;
 /* L140: */
 		}
 	    }
 	} else {
 	    if (*incx == 1) {
 		for (j = *n; j >= 1; --j) {
-		    i__1 = j;
 		    temp.r = X(j).r, temp.i = X(j).i;
 		    if (noconj) {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
-			    i__2 = i + j * a_dim1;
-			    i__3 = i;
 			    q__2.r = A(i,j).r * X(i).r - A(i,j).i * X(
 				    i).i, q__2.i = A(i,j).r * X(i).i + 
 				    A(i,j).i * X(i).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -424,12 +375,10 @@
 			    c_div(&q__1, &temp, &A(j,j));
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    } else {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
 			    r_cnjg(&q__3, &A(i,j));
-			    i__2 = i;
 			    q__2.r = q__3.r * X(i).r - q__3.i * X(i).i, 
 				    q__2.i = q__3.r * X(i).i + q__3.i * X(
 				    i).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -442,24 +391,19 @@
 			    c_div(&q__1, &temp, &q__2);
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    }
-		    i__1 = j;
 		    X(j).r = temp.r, X(j).i = temp.i;
 /* L170: */
 		}
 	    } else {
 		kx += (*n - 1) * *incx;
 		jx = kx;
 		for (j = *n; j >= 1; --j) {
 		    ix = kx;
-		    i__1 = jx;
 		    temp.r = X(jx).r, temp.i = X(jx).i;
 		    if (noconj) {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
-			    i__2 = i + j * a_dim1;
-			    i__3 = ix;
 			    q__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(
 				    ix).i, q__2.i = A(i,j).r * X(ix).i + 
 				    A(i,j).i * X(ix).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -472,12 +416,10 @@
 			    c_div(&q__1, &temp, &A(j,j));
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    } else {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
 			    r_cnjg(&q__3, &A(i,j));
-			    i__2 = ix;
 			    q__2.r = q__3.r * X(ix).r - q__3.i * X(ix).i, 
 				    q__2.i = q__3.r * X(ix).i + q__3.i * X(
 				    ix).r;
 			    q__1.r = temp.r - q__2.r, q__1.i = temp.i - 
@@ -491,9 +433,8 @@
 			    c_div(&q__1, &temp, &q__2);
 			    temp.r = q__1.r, temp.i = q__1.i;
 			}
 		    }
-		    i__1 = jx;
 		    X(jx).r = temp.r, X(jx).i = temp.i;
 		    jx -= *incx;
 /* L200: */
 		}
Index: SuperLU_3.0/CBLAS/zgemv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/zgemv.c
+++ SuperLU_3.0/CBLAS/zgemv.c
@@ -13,9 +13,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublecomplex z__1, z__2, z__3;
 
     /* Builtin functions */
     void d_cnjg(doublecomplex *, doublecomplex *);
@@ -195,19 +194,14 @@
 
     if (beta->r != 1. || beta->i != 0.) {
 	if (*incy == 1) {
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0., Y(i).i = 0.;
 /* L10: */
 		}
 	    } else {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    z__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    z__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
@@ -216,20 +210,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0., Y(iy).i = 0.;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = leny;
 		for (i = 1; i <= leny; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    z__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    z__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
@@ -247,22 +236,15 @@
 /*        Form  y := alpha*A*x + y. */
 
 	jx = kx;
 	if (*incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		if (X(jx).r != 0. || X(jx).i != 0.) {
-		    i__2 = jx;
 		    z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    z__1.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    temp.r = z__1.r, temp.i = z__1.i;
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i;
-			i__4 = i;
-			i__5 = i + j * a_dim1;
 			z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				z__2.i = temp.r * A(i,j).i + temp.i * A(i,j)
 				.r;
 			z__1.r = Y(i).r + z__2.r, z__1.i = Y(i).i + 
@@ -274,23 +256,16 @@
 		jx += *incx;
 /* L60: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		if (X(jx).r != 0. || X(jx).i != 0.) {
-		    i__2 = jx;
 		    z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    z__1.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    temp.r = z__1.r, temp.i = z__1.i;
 		    iy = ky;
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = iy;
-			i__4 = iy;
-			i__5 = i + j * a_dim1;
 			z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				z__2.i = temp.r * A(i,j).i + temp.i * A(i,j)
 				.r;
 			z__1.r = Y(iy).r + z__2.r, z__1.i = Y(iy).i + 
@@ -310,55 +285,43 @@
  */
 
 	jy = ky;
 	if (*incx == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
 		temp.r = 0., temp.i = 0.;
 		if (noconj) {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i;
 			z__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i)
 				.i, z__2.i = A(i,j).r * X(i).i + A(i,j)
 				.i * X(i).r;
 			z__1.r = temp.r + z__2.r, z__1.i = temp.i + z__2.i;
 			temp.r = z__1.r, temp.i = z__1.i;
 /* L90: */
 		    }
 		} else {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
 			d_cnjg(&z__3, &A(i,j));
-			i__3 = i;
 			z__2.r = z__3.r * X(i).r - z__3.i * X(i).i, 
 				z__2.i = z__3.r * X(i).i + z__3.i * X(i)
 				.r;
 			z__1.r = temp.r + z__2.r, z__1.i = temp.i + z__2.i;
 			temp.r = z__1.r, temp.i = z__1.i;
 /* L100: */
 		    }
 		}
-		i__2 = jy;
-		i__3 = jy;
 		z__2.r = alpha->r * temp.r - alpha->i * temp.i, z__2.i = 
 			alpha->r * temp.i + alpha->i * temp.r;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
 		jy += *incy;
 /* L110: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
 		temp.r = 0., temp.i = 0.;
 		ix = kx;
 		if (noconj) {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = ix;
 			z__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix)
 				.i, z__2.i = A(i,j).r * X(ix).i + A(i,j)
 				.i * X(ix).r;
 			z__1.r = temp.r + z__2.r, z__1.i = temp.i + z__2.i;
@@ -366,12 +329,10 @@
 			ix += *incx;
 /* L120: */
 		    }
 		} else {
-		    i__2 = *m;
 		    for (i = 1; i <= *m; ++i) {
 			d_cnjg(&z__3, &A(i,j));
-			i__3 = ix;
 			z__2.r = z__3.r * X(ix).r - z__3.i * X(ix).i, 
 				z__2.i = z__3.r * X(ix).i + z__3.i * X(ix)
 				.r;
 			z__1.r = temp.r + z__2.r, z__1.i = temp.i + z__2.i;
@@ -379,10 +340,8 @@
 			ix += *incx;
 /* L130: */
 		    }
 		}
-		i__2 = jy;
-		i__3 = jy;
 		z__2.r = alpha->r * temp.r - alpha->i * temp.i, z__2.i = 
 			alpha->r * temp.i + alpha->i * temp.r;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
Index: SuperLU_3.0/CBLAS/zgerc.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/zgerc.c
+++ SuperLU_3.0/CBLAS/zgerc.c
@@ -12,9 +12,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublecomplex z__1, z__2;
 
     /* Builtin functions */
     void d_cnjg(doublecomplex *, doublecomplex *);
@@ -141,21 +140,15 @@
     } else {
 	jy = 1 - (*n - 1) * *incy;
     }
     if (*incx == 1) {
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = jy;
 	    if (Y(jy).r != 0. || Y(jy).i != 0.) {
 		d_cnjg(&z__2, &Y(jy));
 		z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			alpha->r * z__2.i + alpha->i * z__2.r;
 		temp.r = z__1.r, temp.i = z__1.i;
-		i__2 = *m;
 		for (i = 1; i <= *m; ++i) {
-		    i__3 = i + j * a_dim1;
-		    i__4 = i + j * a_dim1;
-		    i__5 = i;
 		    z__2.r = X(i).r * temp.r - X(i).i * temp.i, z__2.i =
 			     X(i).r * temp.i + X(i).i * temp.r;
 		    z__1.r = A(i,j).r + z__2.r, z__1.i = A(i,j).i + z__2.i;
 		    A(i,j).r = z__1.r, A(i,j).i = z__1.i;
@@ -170,22 +163,16 @@
 	    kx = 1;
 	} else {
 	    kx = 1 - (*m - 1) * *incx;
 	}
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = jy;
 	    if (Y(jy).r != 0. || Y(jy).i != 0.) {
 		d_cnjg(&z__2, &Y(jy));
 		z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			alpha->r * z__2.i + alpha->i * z__2.r;
 		temp.r = z__1.r, temp.i = z__1.i;
 		ix = kx;
-		i__2 = *m;
 		for (i = 1; i <= *m; ++i) {
-		    i__3 = i + j * a_dim1;
-		    i__4 = i + j * a_dim1;
-		    i__5 = ix;
 		    z__2.r = X(ix).r * temp.r - X(ix).i * temp.i, z__2.i =
 			     X(ix).r * temp.i + X(ix).i * temp.r;
 		    z__1.r = A(i,j).r + z__2.r, z__1.i = A(i,j).i + z__2.i;
 		    A(i,j).r = z__1.r, A(i,j).i = z__1.i;
Index: SuperLU_3.0/CBLAS/zhemv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/zhemv.c
+++ SuperLU_3.0/CBLAS/zhemv.c
@@ -12,9 +12,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublereal d__1;
     doublecomplex z__1, z__2, z__3, z__4;
 
     /* Builtin functions */
@@ -180,19 +179,14 @@
 
     if (beta->r != 1. || beta->i != 0.) {
 	if (*incy == 1) {
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0., Y(i).i = 0.;
 /* L10: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    z__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    z__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
@@ -201,20 +195,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0., Y(iy).i = 0.;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    z__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    z__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
@@ -231,36 +220,26 @@
 
 /*        Form  y  when A is stored in upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		z__1.r = alpha->r * X(j).r - alpha->i * X(j).i, z__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(i).r + z__2.r, z__1.i = Y(i).i + z__2.i;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
 		    d_cnjg(&z__3, &A(i,j));
-		    i__3 = i;
 		    z__2.r = z__3.r * X(i).r - z__3.i * X(i).i, z__2.i =
 			     z__3.r * X(i).i + z__3.i * X(i).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L50: */
 		}
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		z__3.r = d__1 * temp1.r, z__3.i = d__1 * temp1.i;
 		z__2.r = Y(j).r + z__3.r, z__2.i = Y(j).i + z__3.i;
 		z__4.r = alpha->r * temp2.r - alpha->i * temp2.i, z__4.i = 
@@ -271,40 +250,30 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, z__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
 		ix = kx;
 		iy = ky;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(iy).r + z__2.r, z__1.i = Y(iy).i + z__2.i;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
 		    d_cnjg(&z__3, &A(i,j));
-		    i__3 = ix;
 		    z__2.r = z__3.r * X(ix).r - z__3.i * X(ix).i, z__2.i =
 			     z__3.r * X(ix).i + z__3.i * X(ix).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 		    ix += *incx;
 		    iy += *incy;
 /* L70: */
 		}
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		z__3.r = d__1 * temp1.r, z__3.i = d__1 * temp1.i;
 		z__2.r = Y(jy).r + z__3.r, z__2.i = Y(jy).i + z__3.i;
 		z__4.r = alpha->r * temp2.r - alpha->i * temp2.i, z__4.i = 
@@ -320,42 +289,30 @@
 
 /*        Form  y  when A is stored in lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		z__1.r = alpha->r * X(j).r - alpha->i * X(j).i, z__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		z__2.r = d__1 * temp1.r, z__2.i = d__1 * temp1.i;
 		z__1.r = Y(j).r + z__2.r, z__1.i = Y(j).i + z__2.i;
 		Y(j).r = z__1.r, Y(j).i = z__1.i;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(i).r + z__2.r, z__1.i = Y(i).i + z__2.i;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
 		    d_cnjg(&z__3, &A(i,j));
-		    i__3 = i;
 		    z__2.r = z__3.r * X(i).r - z__3.i * X(i).i, z__2.i =
 			     z__3.r * X(i).i + z__3.i * X(i).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L90: */
 		}
-		i__2 = j;
-		i__3 = j;
 		z__2.r = alpha->r * temp2.r - alpha->i * temp2.i, z__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		z__1.r = Y(j).r + z__2.r, z__1.i = Y(j).i + z__2.i;
 		Y(j).r = z__1.r, Y(j).i = z__1.i;
@@ -363,46 +320,34 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, z__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		d__1 = A(j,j).r;
 		z__2.r = d__1 * temp1.r, z__2.i = d__1 * temp1.i;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
 		ix = jx;
 		iy = jy;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
 		    ix += *incx;
 		    iy += *incy;
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(iy).r + z__2.r, z__1.i = Y(iy).i + z__2.i;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
 		    d_cnjg(&z__3, &A(i,j));
-		    i__3 = ix;
 		    z__2.r = z__3.r * X(ix).r - z__3.i * X(ix).i, z__2.i =
 			     z__3.r * X(ix).i + z__3.i * X(ix).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L110: */
 		}
-		i__2 = jy;
-		i__3 = jy;
 		z__2.r = alpha->r * temp2.r - alpha->i * temp2.i, z__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
Index: SuperLU_3.0/CBLAS/zher2.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/zher2.c
+++ SuperLU_3.0/CBLAS/zher2.c
@@ -12,9 +12,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5, i__6;
     doublereal d__1;
     doublecomplex z__1, z__2, z__3, z__4;
 
     /* Builtin functions */
@@ -185,93 +184,69 @@
 
 /*        Form  A  when A is stored in the upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
-		i__3 = j;
 		if (X(j).r != 0. || X(j).i != 0. || (Y(j).r != 0. || 
 			Y(j).i != 0.)) {
 		    d_cnjg(&z__2, &Y(j));
 		    z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			    alpha->r * z__2.i + alpha->i * z__2.r;
 		    temp1.r = z__1.r, temp1.i = z__1.i;
-		    i__2 = j;
 		    z__2.r = alpha->r * X(j).r - alpha->i * X(j).i, 
 			    z__2.i = alpha->r * X(j).i + alpha->i * X(j)
 			    .r;
 		    d_cnjg(&z__1, &z__2);
 		    temp2.r = z__1.r, temp2.i = z__1.i;
-		    i__2 = j - 1;
 		    for (i = 1; i <= j-1; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = i;
 			z__3.r = X(i).r * temp1.r - X(i).i * temp1.i, 
 				z__3.i = X(i).r * temp1.i + X(i).i * 
 				temp1.r;
 			z__2.r = A(i,j).r + z__3.r, z__2.i = A(i,j).i + 
 				z__3.i;
-			i__6 = i;
 			z__4.r = Y(i).r * temp2.r - Y(i).i * temp2.i, 
 				z__4.i = Y(i).r * temp2.i + Y(i).i * 
 				temp2.r;
 			z__1.r = z__2.r + z__4.r, z__1.i = z__2.i + z__4.i;
 			A(i,j).r = z__1.r, A(i,j).i = z__1.i;
 /* L10: */
 		    }
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = j;
 		    z__2.r = X(j).r * temp1.r - X(j).i * temp1.i, 
 			    z__2.i = X(j).r * temp1.i + X(j).i * 
 			    temp1.r;
-		    i__5 = j;
 		    z__3.r = Y(j).r * temp2.r - Y(j).i * temp2.i, 
 			    z__3.i = Y(j).r * temp2.i + Y(j).i * 
 			    temp2.r;
 		    z__1.r = z__2.r + z__3.r, z__1.i = z__2.i + z__3.i;
 		    d__1 = A(j,j).r + z__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		}
 /* L20: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
-		i__3 = jy;
 		if (X(jx).r != 0. || X(jx).i != 0. || (Y(jy).r != 0. || 
 			Y(jy).i != 0.)) {
 		    d_cnjg(&z__2, &Y(jy));
 		    z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			    alpha->r * z__2.i + alpha->i * z__2.r;
 		    temp1.r = z__1.r, temp1.i = z__1.i;
-		    i__2 = jx;
 		    z__2.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    z__2.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    d_cnjg(&z__1, &z__2);
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 		    ix = kx;
 		    iy = ky;
-		    i__2 = j - 1;
 		    for (i = 1; i <= j-1; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = ix;
 			z__3.r = X(ix).r * temp1.r - X(ix).i * temp1.i, 
 				z__3.i = X(ix).r * temp1.i + X(ix).i * 
 				temp1.r;
 			z__2.r = A(i,j).r + z__3.r, z__2.i = A(i,j).i + 
 				z__3.i;
-			i__6 = iy;
 			z__4.r = Y(iy).r * temp2.r - Y(iy).i * temp2.i, 
 				z__4.i = Y(iy).r * temp2.i + Y(iy).i * 
 				temp2.r;
 			z__1.r = z__2.r + z__4.r, z__1.i = z__2.i + z__4.i;
@@ -279,24 +254,18 @@
 			ix += *incx;
 			iy += *incy;
 /* L30: */
 		    }
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = jx;
 		    z__2.r = X(jx).r * temp1.r - X(jx).i * temp1.i, 
 			    z__2.i = X(jx).r * temp1.i + X(jx).i * 
 			    temp1.r;
-		    i__5 = jy;
 		    z__3.r = Y(jy).r * temp2.r - Y(jy).i * temp2.i, 
 			    z__3.i = Y(jy).r * temp2.i + Y(jy).i * 
 			    temp2.r;
 		    z__1.r = z__2.r + z__3.r, z__1.i = z__2.i + z__3.i;
 		    d__1 = A(j,j).r + z__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		}
 		jx += *incx;
@@ -308,118 +277,88 @@
 
 /*        Form  A  when A is stored in the lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
-		i__3 = j;
 		if (X(j).r != 0. || X(j).i != 0. || (Y(j).r != 0. || 
 			Y(j).i != 0.)) {
 		    d_cnjg(&z__2, &Y(j));
 		    z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			    alpha->r * z__2.i + alpha->i * z__2.r;
 		    temp1.r = z__1.r, temp1.i = z__1.i;
-		    i__2 = j;
 		    z__2.r = alpha->r * X(j).r - alpha->i * X(j).i, 
 			    z__2.i = alpha->r * X(j).i + alpha->i * X(j)
 			    .r;
 		    d_cnjg(&z__1, &z__2);
 		    temp2.r = z__1.r, temp2.i = z__1.i;
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = j;
 		    z__2.r = X(j).r * temp1.r - X(j).i * temp1.i, 
 			    z__2.i = X(j).r * temp1.i + X(j).i * 
 			    temp1.r;
-		    i__5 = j;
 		    z__3.r = Y(j).r * temp2.r - Y(j).i * temp2.i, 
 			    z__3.i = Y(j).r * temp2.i + Y(j).i * 
 			    temp2.r;
 		    z__1.r = z__2.r + z__3.r, z__1.i = z__2.i + z__3.i;
 		    d__1 = A(j,j).r + z__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
-		    i__2 = *n;
 		    for (i = j + 1; i <= *n; ++i) {
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = i;
 			z__3.r = X(i).r * temp1.r - X(i).i * temp1.i, 
 				z__3.i = X(i).r * temp1.i + X(i).i * 
 				temp1.r;
 			z__2.r = A(i,j).r + z__3.r, z__2.i = A(i,j).i + 
 				z__3.i;
-			i__6 = i;
 			z__4.r = Y(i).r * temp2.r - Y(i).i * temp2.i, 
 				z__4.i = Y(i).r * temp2.i + Y(i).i * 
 				temp2.r;
 			z__1.r = z__2.r + z__4.r, z__1.i = z__2.i + z__4.i;
 			A(i,j).r = z__1.r, A(i,j).i = z__1.i;
 /* L50: */
 		    }
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		}
 /* L60: */
 	    }
 	} else {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
-		i__3 = jy;
 		if (X(jx).r != 0. || X(jx).i != 0. || (Y(jy).r != 0. || 
 			Y(jy).i != 0.)) {
 		    d_cnjg(&z__2, &Y(jy));
 		    z__1.r = alpha->r * z__2.r - alpha->i * z__2.i, z__1.i = 
 			    alpha->r * z__2.i + alpha->i * z__2.r;
 		    temp1.r = z__1.r, temp1.i = z__1.i;
-		    i__2 = jx;
 		    z__2.r = alpha->r * X(jx).r - alpha->i * X(jx).i, 
 			    z__2.i = alpha->r * X(jx).i + alpha->i * X(jx)
 			    .r;
 		    d_cnjg(&z__1, &z__2);
 		    temp2.r = z__1.r, temp2.i = z__1.i;
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
-		    i__4 = jx;
 		    z__2.r = X(jx).r * temp1.r - X(jx).i * temp1.i, 
 			    z__2.i = X(jx).r * temp1.i + X(jx).i * 
 			    temp1.r;
-		    i__5 = jy;
 		    z__3.r = Y(jy).r * temp2.r - Y(jy).i * temp2.i, 
 			    z__3.i = Y(jy).r * temp2.i + Y(jy).i * 
 			    temp2.r;
 		    z__1.r = z__2.r + z__3.r, z__1.i = z__2.i + z__3.i;
 		    d__1 = A(j,j).r + z__1.r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		    ix = jx;
 		    iy = jy;
-		    i__2 = *n;
 		    for (i = j + 1; i <= *n; ++i) {
 			ix += *incx;
 			iy += *incy;
-			i__3 = i + j * a_dim1;
-			i__4 = i + j * a_dim1;
-			i__5 = ix;
 			z__3.r = X(ix).r * temp1.r - X(ix).i * temp1.i, 
 				z__3.i = X(ix).r * temp1.i + X(ix).i * 
 				temp1.r;
 			z__2.r = A(i,j).r + z__3.r, z__2.i = A(i,j).i + 
 				z__3.i;
-			i__6 = iy;
 			z__4.r = Y(iy).r * temp2.r - Y(iy).i * temp2.i, 
 				z__4.i = Y(iy).r * temp2.i + Y(iy).i * 
 				temp2.r;
 			z__1.r = z__2.r + z__4.r, z__1.i = z__2.i + z__4.i;
 			A(i,j).r = z__1.r, A(i,j).i = z__1.i;
 /* L70: */
 		    }
 		} else {
-		    i__2 = j + j * a_dim1;
-		    i__3 = j + j * a_dim1;
 		    d__1 = A(j,j).r;
 		    A(j,j).r = d__1, A(j,j).i = 0.;
 		}
 		jx += *incx;
Index: SuperLU_3.0/CBLAS/ztrsv.c
===================================================================
--- SuperLU_3.0.orig/CBLAS/ztrsv.c
+++ SuperLU_3.0/CBLAS/ztrsv.c
@@ -11,9 +11,8 @@
 {
 
 
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublecomplex z__1, z__2, z__3;
 
     /* Builtin functions */
     void z_div(doublecomplex *, doublecomplex *, doublecomplex *), d_cnjg(
@@ -187,21 +186,15 @@
 
 	if (lsame_(uplo, "U")) {
 	    if (*incx == 1) {
 		for (j = *n; j >= 1; --j) {
-		    i__1 = j;
 		    if (X(j).r != 0. || X(j).i != 0.) {
 			if (nounit) {
-			    i__1 = j;
 			    z_div(&z__1, &X(j), &A(j,j));
 			    X(j).r = z__1.r, X(j).i = z__1.i;
 			}
-			i__1 = j;
 			temp.r = X(j).r, temp.i = X(j).i;
 			for (i = j - 1; i >= 1; --i) {
-			    i__1 = i;
-			    i__2 = i;
-			    i__3 = i + j * a_dim1;
 			    z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    z__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    z__1.r = X(i).r - z__2.r, z__1.i = X(i).i - 
 				    z__2.i;
@@ -213,23 +206,17 @@
 		}
 	    } else {
 		jx = kx + (*n - 1) * *incx;
 		for (j = *n; j >= 1; --j) {
-		    i__1 = jx;
 		    if (X(jx).r != 0. || X(jx).i != 0.) {
 			if (nounit) {
-			    i__1 = jx;
 			    z_div(&z__1, &X(jx), &A(j,j));
 			    X(jx).r = z__1.r, X(jx).i = z__1.i;
 			}
-			i__1 = jx;
 			temp.r = X(jx).r, temp.i = X(jx).i;
 			ix = jx;
 			for (i = j - 1; i >= 1; --i) {
 			    ix -= *incx;
-			    i__1 = ix;
-			    i__2 = ix;
-			    i__3 = i + j * a_dim1;
 			    z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    z__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    z__1.r = X(ix).r - z__2.r, z__1.i = X(ix).i - 
 				    z__2.i;
@@ -242,24 +229,16 @@
 		}
 	    }
 	} else {
 	    if (*incx == 1) {
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = j;
 		    if (X(j).r != 0. || X(j).i != 0.) {
 			if (nounit) {
-			    i__2 = j;
 			    z_div(&z__1, &X(j), &A(j,j));
 			    X(j).r = z__1.r, X(j).i = z__1.i;
 			}
-			i__2 = j;
 			temp.r = X(j).r, temp.i = X(j).i;
-			i__2 = *n;
 			for (i = j + 1; i <= *n; ++i) {
-			    i__3 = i;
-			    i__4 = i;
-			    i__5 = i + j * a_dim1;
 			    z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    z__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    z__1.r = X(i).r - z__2.r, z__1.i = X(i).i - 
 				    z__2.i;
@@ -270,26 +249,18 @@
 /* L60: */
 		}
 	    } else {
 		jx = kx;
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = jx;
 		    if (X(jx).r != 0. || X(jx).i != 0.) {
 			if (nounit) {
-			    i__2 = jx;
 			    z_div(&z__1, &X(jx), &A(j,j));
 			    X(jx).r = z__1.r, X(jx).i = z__1.i;
 			}
-			i__2 = jx;
 			temp.r = X(jx).r, temp.i = X(jx).i;
 			ix = jx;
-			i__2 = *n;
 			for (i = j + 1; i <= *n; ++i) {
 			    ix += *incx;
-			    i__3 = ix;
-			    i__4 = ix;
-			    i__5 = i + j * a_dim1;
 			    z__2.r = temp.r * A(i,j).r - temp.i * A(i,j).i, 
 				    z__2.i = temp.r * A(i,j).i + temp.i * A(i,j).r;
 			    z__1.r = X(ix).r - z__2.r, z__1.i = X(ix).i - 
 				    z__2.i;
@@ -307,17 +278,12 @@
 /*        Form  x := inv( A' )*x  or  x := inv( conjg( A' ) )*x. */
 
 	if (lsame_(uplo, "U")) {
 	    if (*incx == 1) {
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
-		    i__2 = j;
 		    temp.r = X(j).r, temp.i = X(j).i;
 		    if (noconj) {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
-			    i__3 = i + j * a_dim1;
-			    i__4 = i;
 			    z__2.r = A(i,j).r * X(i).r - A(i,j).i * X(
 				    i).i, z__2.i = A(i,j).r * X(i).i + 
 				    A(i,j).i * X(i).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -329,12 +295,10 @@
 			    z_div(&z__1, &temp, &A(j,j));
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    } else {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
 			    d_cnjg(&z__3, &A(i,j));
-			    i__3 = i;
 			    z__2.r = z__3.r * X(i).r - z__3.i * X(i).i, 
 				    z__2.i = z__3.r * X(i).i + z__3.i * X(
 				    i).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -347,24 +311,18 @@
 			    z_div(&z__1, &temp, &z__2);
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    }
-		    i__2 = j;
 		    X(j).r = temp.r, X(j).i = temp.i;
 /* L110: */
 		}
 	    } else {
 		jx = kx;
-		i__1 = *n;
 		for (j = 1; j <= *n; ++j) {
 		    ix = kx;
-		    i__2 = jx;
 		    temp.r = X(jx).r, temp.i = X(jx).i;
 		    if (noconj) {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
-			    i__3 = i + j * a_dim1;
-			    i__4 = ix;
 			    z__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(
 				    ix).i, z__2.i = A(i,j).r * X(ix).i + 
 				    A(i,j).i * X(ix).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -377,12 +335,10 @@
 			    z_div(&z__1, &temp, &A(j,j));
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    } else {
-			i__2 = j - 1;
 			for (i = 1; i <= j-1; ++i) {
 			    d_cnjg(&z__3, &A(i,j));
-			    i__3 = ix;
 			    z__2.r = z__3.r * X(ix).r - z__3.i * X(ix).i, 
 				    z__2.i = z__3.r * X(ix).i + z__3.i * X(
 				    ix).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -396,24 +352,19 @@
 			    z_div(&z__1, &temp, &z__2);
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    }
-		    i__2 = jx;
 		    X(jx).r = temp.r, X(jx).i = temp.i;
 		    jx += *incx;
 /* L140: */
 		}
 	    }
 	} else {
 	    if (*incx == 1) {
 		for (j = *n; j >= 1; --j) {
-		    i__1 = j;
 		    temp.r = X(j).r, temp.i = X(j).i;
 		    if (noconj) {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
-			    i__2 = i + j * a_dim1;
-			    i__3 = i;
 			    z__2.r = A(i,j).r * X(i).r - A(i,j).i * X(
 				    i).i, z__2.i = A(i,j).r * X(i).i + 
 				    A(i,j).i * X(i).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -425,12 +376,10 @@
 			    z_div(&z__1, &temp, &A(j,j));
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    } else {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
 			    d_cnjg(&z__3, &A(i,j));
-			    i__2 = i;
 			    z__2.r = z__3.r * X(i).r - z__3.i * X(i).i, 
 				    z__2.i = z__3.r * X(i).i + z__3.i * X(
 				    i).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -443,24 +392,19 @@
 			    z_div(&z__1, &temp, &z__2);
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    }
-		    i__1 = j;
 		    X(j).r = temp.r, X(j).i = temp.i;
 /* L170: */
 		}
 	    } else {
 		kx += (*n - 1) * *incx;
 		jx = kx;
 		for (j = *n; j >= 1; --j) {
 		    ix = kx;
-		    i__1 = jx;
 		    temp.r = X(jx).r, temp.i = X(jx).i;
 		    if (noconj) {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
-			    i__2 = i + j * a_dim1;
-			    i__3 = ix;
 			    z__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(
 				    ix).i, z__2.i = A(i,j).r * X(ix).i + 
 				    A(i,j).i * X(ix).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -473,12 +417,10 @@
 			    z_div(&z__1, &temp, &A(j,j));
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    } else {
-			i__1 = j + 1;
 			for (i = *n; i >= j+1; --i) {
 			    d_cnjg(&z__3, &A(i,j));
-			    i__2 = ix;
 			    z__2.r = z__3.r * X(ix).r - z__3.i * X(ix).i, 
 				    z__2.i = z__3.r * X(ix).i + z__3.i * X(
 				    ix).r;
 			    z__1.r = temp.r - z__2.r, z__1.i = temp.i - 
@@ -492,9 +434,8 @@
 			    z_div(&z__1, &temp, &z__2);
 			    temp.r = z__1.r, temp.i = z__1.i;
 			}
 		    }
-		    i__1 = jx;
 		    X(jx).r = temp.r, X(jx).i = temp.i;
 		    jx -= *incx;
 /* L200: */
 		}
Index: SuperLU_3.0/SRC/sutil.c
===================================================================
--- SuperLU_3.0.orig/SRC/sutil.c
+++ SuperLU_3.0/SRC/sutil.c
@@ -266,14 +266,15 @@ sPrint_SuperNode_Matrix(char *what, Supe
 void
 sPrint_Dense_Matrix(char *what, SuperMatrix *A)
 {
     DNformat     *Astore;
-    register int i, j, lda = Astore->lda;
+    register int i, j, lda;
     float       *dp;
     
     printf("\nDense matrix %s:\n", what);
     printf("Stype %d, Dtype %d, Mtype %d\n", A->Stype,A->Dtype,A->Mtype);
     Astore = (DNformat *) A->Store;
+    lda = Astore->lda;
     dp = (float *) Astore->nzval;
     printf("nrow %d, ncol %d, lda %d\n", A->nrow,A->ncol,lda);
     printf("\nnzval: ");
     for (j = 0; j < A->ncol; ++j) {
Index: SuperLU_3.0/SRC/cutil.c
===================================================================
--- SuperLU_3.0.orig/SRC/cutil.c
+++ SuperLU_3.0/SRC/cutil.c
@@ -266,14 +266,15 @@ cPrint_SuperNode_Matrix(char *what, Supe
 void
 cPrint_Dense_Matrix(char *what, SuperMatrix *A)
 {
     DNformat     *Astore;
-    register int i, j, lda = Astore->lda;
+    register int i, j, lda;
     float       *dp;
     
     printf("\nDense matrix %s:\n", what);
     printf("Stype %d, Dtype %d, Mtype %d\n", A->Stype,A->Dtype,A->Mtype);
     Astore = (DNformat *) A->Store;
+    lda = Astore->lda;
     dp = (float *) Astore->nzval;
     printf("nrow %d, ncol %d, lda %d\n", A->nrow,A->ncol,lda);
     printf("\nnzval: ");
     for (j = 0; j < A->ncol; ++j) {
Index: SuperLU_3.0/SRC/dutil.c
===================================================================
--- SuperLU_3.0.orig/SRC/dutil.c
+++ SuperLU_3.0/SRC/dutil.c
@@ -266,14 +266,15 @@ dPrint_SuperNode_Matrix(char *what, Supe
 void
 dPrint_Dense_Matrix(char *what, SuperMatrix *A)
 {
     DNformat     *Astore;
-    register int i, j, lda = Astore->lda;
+    register int i, j, lda;
     double       *dp;
     
     printf("\nDense matrix %s:\n", what);
     printf("Stype %d, Dtype %d, Mtype %d\n", A->Stype,A->Dtype,A->Mtype);
     Astore = (DNformat *) A->Store;
+    lda = Astore->lda;
     dp = (double *) Astore->nzval;
     printf("nrow %d, ncol %d, lda %d\n", A->nrow,A->ncol,lda);
     printf("\nnzval: ");
     for (j = 0; j < A->ncol; ++j) {
Index: SuperLU_3.0/SRC/zutil.c
===================================================================
--- SuperLU_3.0.orig/SRC/zutil.c
+++ SuperLU_3.0/SRC/zutil.c
@@ -266,14 +266,15 @@ zPrint_SuperNode_Matrix(char *what, Supe
 void
 zPrint_Dense_Matrix(char *what, SuperMatrix *A)
 {
     DNformat     *Astore;
-    register int i, j, lda = Astore->lda;
+    register int i, j, lda;
     double       *dp;
     
     printf("\nDense matrix %s:\n", what);
     printf("Stype %d, Dtype %d, Mtype %d\n", A->Stype,A->Dtype,A->Mtype);
     Astore = (DNformat *) A->Store;
+    lda = Astore->lda;
     dp = (double *) Astore->nzval;
     printf("nrow %d, ncol %d, lda %d\n", A->nrow,A->ncol,lda);
     printf("\nnzval: ");
     for (j = 0; j < A->ncol; ++j) {
Index: SuperLU_3.0/TESTING/MATGEN/claset.c
===================================================================
--- SuperLU_3.0.orig/TESTING/MATGEN/claset.c
+++ SuperLU_3.0/TESTING/MATGEN/claset.c
@@ -55,9 +55,8 @@
     
    Parameter adjustments   
        Function Body */
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3;
     /* Local variables */
     static integer i, j;
     extern logical lsame_(char *, char *);
 
@@ -70,23 +69,17 @@
 /*        Set the diagonal to BETA and the strictly upper triangular 
   
           part of the array to ALPHA. */
 
-	i__1 = *n;
 	for (j = 2; j <= *n; ++j) {
 /* Computing MIN */
-	    i__3 = j - 1;
-	    i__2 = min(i__3,*m);
 	    for (i = 1; i <= min(j-1,*m); ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L10: */
 	    }
 /* L20: */
 	}
-	i__1 = min(*n,*m);
 	for (i = 1; i <= min(*n,*m); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L30: */
 	}
 
@@ -95,21 +88,16 @@
 /*        Set the diagonal to BETA and the strictly lower triangular 
   
           part of the array to ALPHA. */
 
-	i__1 = min(*m,*n);
 	for (j = 1; j <= min(*m,*n); ++j) {
-	    i__2 = *m;
 	    for (i = j + 1; i <= *m; ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L40: */
 	    }
 /* L50: */
 	}
-	i__1 = min(*n,*m);
 	for (i = 1; i <= min(*n,*m); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L60: */
 	}
 
@@ -117,21 +105,16 @@
 
 /*        Set the array to BETA on the diagonal and ALPHA on the   
           offdiagonal. */
 
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = *m;
 	    for (i = 1; i <= *m; ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L70: */
 	    }
 /* L80: */
 	}
-	i__1 = min(*m,*n);
 	for (i = 1; i <= min(*m,*n); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L90: */
 	}
     }
Index: SuperLU_3.0/TESTING/MATGEN/csymv.c
===================================================================
--- SuperLU_3.0.orig/TESTING/MATGEN/csymv.c
+++ SuperLU_3.0/TESTING/MATGEN/csymv.c
@@ -101,9 +101,8 @@
     
    Parameter adjustments   
        Function Body */
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     complex q__1, q__2, q__3, q__4;
     /* Local variables */
     static integer info;
     static complex temp1, temp2;
@@ -163,19 +162,14 @@
 
     if (beta->r != 1.f || beta->i != 0.f) {
 	if (*incy == 1) {
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0.f, Y(i).i = 0.f;
 /* L10: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    q__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    q__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
@@ -184,20 +178,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0.f && beta->i == 0.f) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0.f, Y(iy).i = 0.f;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    q__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    q__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
@@ -214,37 +203,26 @@
 
 /*        Form  y  when A is stored in upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		q__1.r = alpha->r * X(j).r - alpha->i * X(j).i, q__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(i).r + q__2.r, q__1.i = Y(i).i + q__2.i;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = i;
 		    q__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i).i, 
 			    q__2.i = A(i,j).r * X(i).i + A(i,j).i * X(
 			    i).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L50: */
 		}
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		q__3.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, q__3.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		q__2.r = Y(j).r + q__3.r, q__2.i = Y(j).i + q__3.i;
 		q__4.r = alpha->r * temp2.r - alpha->i * temp2.i, q__4.i = 
@@ -255,29 +233,21 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, q__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
 		ix = kx;
 		iy = ky;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(iy).r + q__2.r, q__1.i = Y(iy).i + q__2.i;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = ix;
 		    q__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix).i, 
 			    q__2.i = A(i,j).r * X(ix).i + A(i,j).i * X(
 			    ix).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
@@ -285,11 +255,8 @@
 		    ix += *incx;
 		    iy += *incy;
 /* L70: */
 		}
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		q__3.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, q__3.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		q__2.r = Y(jy).r + q__3.r, q__2.i = Y(jy).i + q__3.i;
 		q__4.r = alpha->r * temp2.r - alpha->i * temp2.i, q__4.i = 
@@ -305,43 +272,30 @@
 
 /*        Form  y  when A is stored in lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		q__1.r = alpha->r * X(j).r - alpha->i * X(j).i, q__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		q__2.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, q__2.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		q__1.r = Y(j).r + q__2.r, q__1.i = Y(j).i + q__2.i;
 		Y(j).r = q__1.r, Y(j).i = q__1.i;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(i).r + q__2.r, q__1.i = Y(i).i + q__2.i;
 		    Y(i).r = q__1.r, Y(i).i = q__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = i;
 		    q__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i).i, 
 			    q__2.i = A(i,j).r * X(i).i + A(i,j).i * X(
 			    i).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L90: */
 		}
-		i__2 = j;
-		i__3 = j;
 		q__2.r = alpha->r * temp2.r - alpha->i * temp2.i, q__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		q__1.r = Y(j).r + q__2.r, q__1.i = Y(j).i + q__2.i;
 		Y(j).r = q__1.r, Y(j).i = q__1.i;
@@ -349,47 +303,34 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		q__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, q__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = q__1.r, temp1.i = q__1.i;
 		temp2.r = 0.f, temp2.i = 0.f;
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		q__2.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, q__2.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
 		ix = jx;
 		iy = jy;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
 		    ix += *incx;
 		    iy += *incy;
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    q__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    q__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    q__1.r = Y(iy).r + q__2.r, q__1.i = Y(iy).i + q__2.i;
 		    Y(iy).r = q__1.r, Y(iy).i = q__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = ix;
 		    q__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix).i, 
 			    q__2.i = A(i,j).r * X(ix).i + A(i,j).i * X(
 			    ix).r;
 		    q__1.r = temp2.r + q__2.r, q__1.i = temp2.i + q__2.i;
 		    temp2.r = q__1.r, temp2.i = q__1.i;
 /* L110: */
 		}
-		i__2 = jy;
-		i__3 = jy;
 		q__2.r = alpha->r * temp2.r - alpha->i * temp2.i, q__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		q__1.r = Y(jy).r + q__2.r, q__1.i = Y(jy).i + q__2.i;
 		Y(jy).r = q__1.r, Y(jy).i = q__1.i;
Index: SuperLU_3.0/TESTING/MATGEN/zlaset.c
===================================================================
--- SuperLU_3.0.orig/TESTING/MATGEN/zlaset.c
+++ SuperLU_3.0/TESTING/MATGEN/zlaset.c
@@ -56,9 +56,8 @@
     
    Parameter adjustments   
        Function Body */
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3;
     /* Local variables */
     static integer i, j;
     extern logical lsame_(char *, char *);
 
@@ -71,23 +70,17 @@
 /*        Set the diagonal to BETA and the strictly upper triangular 
   
           part of the array to ALPHA. */
 
-	i__1 = *n;
 	for (j = 2; j <= *n; ++j) {
 /* Computing MIN */
-	    i__3 = j - 1;
-	    i__2 = min(i__3,*m);
 	    for (i = 1; i <= min(j-1,*m); ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L10: */
 	    }
 /* L20: */
 	}
-	i__1 = min(*n,*m);
 	for (i = 1; i <= min(*n,*m); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L30: */
 	}
 
@@ -96,21 +89,16 @@
 /*        Set the diagonal to BETA and the strictly lower triangular 
   
           part of the array to ALPHA. */
 
-	i__1 = min(*m,*n);
 	for (j = 1; j <= min(*m,*n); ++j) {
-	    i__2 = *m;
 	    for (i = j + 1; i <= *m; ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L40: */
 	    }
 /* L50: */
 	}
-	i__1 = min(*n,*m);
 	for (i = 1; i <= min(*n,*m); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L60: */
 	}
 
@@ -118,21 +106,16 @@
 
 /*        Set the array to BETA on the diagonal and ALPHA on the   
           offdiagonal. */
 
-	i__1 = *n;
 	for (j = 1; j <= *n; ++j) {
-	    i__2 = *m;
 	    for (i = 1; i <= *m; ++i) {
-		i__3 = i + j * a_dim1;
 		A(i,j).r = alpha->r, A(i,j).i = alpha->i;
 /* L70: */
 	    }
 /* L80: */
 	}
-	i__1 = min(*m,*n);
 	for (i = 1; i <= min(*m,*n); ++i) {
-	    i__2 = i + i * a_dim1;
 	    A(i,i).r = beta->r, A(i,i).i = beta->i;
 /* L90: */
 	}
     }
Index: SuperLU_3.0/TESTING/MATGEN/zsymv.c
===================================================================
--- SuperLU_3.0.orig/TESTING/MATGEN/zsymv.c
+++ SuperLU_3.0/TESTING/MATGEN/zsymv.c
@@ -101,9 +101,8 @@
     
    Parameter adjustments   
        Function Body */
     /* System generated locals */
-    integer a_dim1, a_offset, i__1, i__2, i__3, i__4, i__5;
     doublecomplex z__1, z__2, z__3, z__4;
     /* Local variables */
     static integer info;
     static doublecomplex temp1, temp2;
@@ -163,19 +162,14 @@
 
     if (beta->r != 1. || beta->i != 0.) {
 	if (*incy == 1) {
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
 		    Y(i).r = 0., Y(i).i = 0.;
 /* L10: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = i;
-		    i__3 = i;
 		    z__1.r = beta->r * Y(i).r - beta->i * Y(i).i, 
 			    z__1.i = beta->r * Y(i).i + beta->i * Y(i)
 			    .r;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
@@ -184,20 +178,15 @@
 	    }
 	} else {
 	    iy = ky;
 	    if (beta->r == 0. && beta->i == 0.) {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
 		    Y(iy).r = 0., Y(iy).i = 0.;
 		    iy += *incy;
 /* L30: */
 		}
 	    } else {
-		i__1 = *n;
 		for (i = 1; i <= *n; ++i) {
-		    i__2 = iy;
-		    i__3 = iy;
 		    z__1.r = beta->r * Y(iy).r - beta->i * Y(iy).i, 
 			    z__1.i = beta->r * Y(iy).i + beta->i * Y(iy)
 			    .r;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
@@ -214,37 +203,26 @@
 
 /*        Form  y  when A is stored in upper triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		z__1.r = alpha->r * X(j).r - alpha->i * X(j).i, z__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(i).r + z__2.r, z__1.i = Y(i).i + z__2.i;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = i;
 		    z__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i).i, 
 			    z__2.i = A(i,j).r * X(i).i + A(i,j).i * X(
 			    i).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L50: */
 		}
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		z__3.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, z__3.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		z__2.r = Y(j).r + z__3.r, z__2.i = Y(j).i + z__3.i;
 		z__4.r = alpha->r * temp2.r - alpha->i * temp2.i, z__4.i = 
@@ -255,29 +233,21 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, z__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
 		ix = kx;
 		iy = ky;
-		i__2 = j - 1;
 		for (i = 1; i <= j-1; ++i) {
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(iy).r + z__2.r, z__1.i = Y(iy).i + z__2.i;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = ix;
 		    z__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix).i, 
 			    z__2.i = A(i,j).r * X(ix).i + A(i,j).i * X(
 			    ix).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
@@ -285,11 +255,8 @@
 		    ix += *incx;
 		    iy += *incy;
 /* L70: */
 		}
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		z__3.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, z__3.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		z__2.r = Y(jy).r + z__3.r, z__2.i = Y(jy).i + z__3.i;
 		z__4.r = alpha->r * temp2.r - alpha->i * temp2.i, z__4.i = 
@@ -305,43 +272,30 @@
 
 /*        Form  y  when A is stored in lower triangle. */
 
 	if (*incx == 1 && *incy == 1) {
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = j;
 		z__1.r = alpha->r * X(j).r - alpha->i * X(j).i, z__1.i =
 			 alpha->r * X(j).i + alpha->i * X(j).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = j;
-		i__3 = j;
-		i__4 = j + j * a_dim1;
 		z__2.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, z__2.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		z__1.r = Y(j).r + z__2.r, z__1.i = Y(j).i + z__2.i;
 		Y(j).r = z__1.r, Y(j).i = z__1.i;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
-		    i__3 = i;
-		    i__4 = i;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(i).r + z__2.r, z__1.i = Y(i).i + z__2.i;
 		    Y(i).r = z__1.r, Y(i).i = z__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = i;
 		    z__2.r = A(i,j).r * X(i).r - A(i,j).i * X(i).i, 
 			    z__2.i = A(i,j).r * X(i).i + A(i,j).i * X(
 			    i).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L90: */
 		}
-		i__2 = j;
-		i__3 = j;
 		z__2.r = alpha->r * temp2.r - alpha->i * temp2.i, z__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		z__1.r = Y(j).r + z__2.r, z__1.i = Y(j).i + z__2.i;
 		Y(j).r = z__1.r, Y(j).i = z__1.i;
@@ -349,47 +303,34 @@
 	    }
 	} else {
 	    jx = kx;
 	    jy = ky;
-	    i__1 = *n;
 	    for (j = 1; j <= *n; ++j) {
-		i__2 = jx;
 		z__1.r = alpha->r * X(jx).r - alpha->i * X(jx).i, z__1.i =
 			 alpha->r * X(jx).i + alpha->i * X(jx).r;
 		temp1.r = z__1.r, temp1.i = z__1.i;
 		temp2.r = 0., temp2.i = 0.;
-		i__2 = jy;
-		i__3 = jy;
-		i__4 = j + j * a_dim1;
 		z__2.r = temp1.r * A(j,j).r - temp1.i * A(j,j).i, z__2.i = 
 			temp1.r * A(j,j).i + temp1.i * A(j,j).r;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
 		ix = jx;
 		iy = jy;
-		i__2 = *n;
 		for (i = j + 1; i <= *n; ++i) {
 		    ix += *incx;
 		    iy += *incy;
-		    i__3 = iy;
-		    i__4 = iy;
-		    i__5 = i + j * a_dim1;
 		    z__2.r = temp1.r * A(i,j).r - temp1.i * A(i,j).i, 
 			    z__2.i = temp1.r * A(i,j).i + temp1.i * A(i,j)
 			    .r;
 		    z__1.r = Y(iy).r + z__2.r, z__1.i = Y(iy).i + z__2.i;
 		    Y(iy).r = z__1.r, Y(iy).i = z__1.i;
-		    i__3 = i + j * a_dim1;
-		    i__4 = ix;
 		    z__2.r = A(i,j).r * X(ix).r - A(i,j).i * X(ix).i, 
 			    z__2.i = A(i,j).r * X(ix).i + A(i,j).i * X(
 			    ix).r;
 		    z__1.r = temp2.r + z__2.r, z__1.i = temp2.i + z__2.i;
 		    temp2.r = z__1.r, temp2.i = z__1.i;
 /* L110: */
 		}
-		i__2 = jy;
-		i__3 = jy;
 		z__2.r = alpha->r * temp2.r - alpha->i * temp2.i, z__2.i = 
 			alpha->r * temp2.i + alpha->i * temp2.r;
 		z__1.r = Y(jy).r + z__2.r, z__1.i = Y(jy).i + z__2.i;
 		Y(jy).r = z__1.r, Y(jy).i = z__1.i;
