diff --git a/SRC/Makefile b/SRC/Makefile
index 2dbd9e5..44a1f87 100644
--- a/SRC/Makefile
+++ b/SRC/Makefile
@@ -91,7 +91,7 @@ ZLUSRC = \
 	ilu_zcolumn_dfs.o ilu_zpanel_dfs.o ilu_zcopy_to_ucol.o \
 	ilu_zpivotL.o zdiagonal.o zlacon2.o dzsum1.o izmax1.o
 
-all:    single double complex complex16
+all:    sharedlib
 
 single: $(SLUSRC) $(ALLAUX)
 	$(ARCH) $(ARCHFLAGS) $(SUPERLULIB) $(SLUSRC) $(ALLAUX)
@@ -109,10 +109,16 @@ complex16: $(ZLUSRC) $(ALLAUX)
 	$(ARCH) $(ARCHFLAGS) $(SUPERLULIB) $(ZLUSRC) $(ALLAUX)
 	$(RANLIB) $(SUPERLULIB)
 
+sharedlib: $(ALLAUX) $(LAAUX) $(SLASRC) $(DLASRC) $(CLASRC) $(ZLASRC) $(SCLAUX) $(DZLAUX) $(SLUSRC) $(DLUSRC) $(CLUSRC) $(ZLUSRC)
+	$(CC) $(CFLAGS) $(LIBS) -shared -Wl,-soname,libsuperlu.so.5.2 -o libsuperlu.so.5.2 \
+	$(ALLAUX) $(LAAUX) $(SLASRC) $(DLASRC) $(CLASRC) $(ZLASRC) $(SCLAUX) \
+	$(DZLAUX) $(SLUSRC) $(DLUSRC) $(CLUSRC) $(ZLUSRC)
+	ln -sf libsuperlu.so.5.2 libsuperlu.so
+
 ##################################
 
 .c.o:
-	$(CC) $(CFLAGS) $(CDEFS) $(BLASDEF) -c $< $(VERBOSE)
+	$(CC) $(CFLAGS) $(CDEFS) $(BLASDEF) $(LIBS) -c $< $(VERBOSE)
 
 .f.o:
 	$(FORTRAN) $(FFLAGS) -c $<
